File: fibonacci/recursive.rb
# Runtime 48 ms, Memory Usage 9.4 MB

def fib(n)
    # create store array
    # this is where we will store the sums we get from the recursive
    store = [0, 1]

    # declare a counter
    # this will help to determine the amount of times we use recursion
    i = 1
    
    # define a recursive method
    def recursive(number, check, store)
        # add the last two elements in the array
        sum = store[-1] + store[-2]

        # push the sum into the store
        store << sum

        # increase the counter
        number += 1

        # if the counter is lower than the check, run the method again
        if number < check
            recursive(number, check, store)
        end
    end
    
    # run the recursive method
    recursive(i, n, store)

    # return the last element in the store
    store[-1]
end

File: fibonacci/recursive.py
# Runtime 32 ms, Memory Usage 11.7 MB

class Solution(object):
    def fib(self, N):
        # declare a store list that will contain all of our fibonacci numbers
        store = [0, 1]
        
        # define a recursive function
        # the arguments will be our store list and a counter
        def recursive(lst, counter):
            # declare a variable that is equal to the sum of the last two elements in the list
            sum = lst[-1] + lst[-2]

            # append that variable to our list
            lst.append(sum)

            if len(lst) <= counter:
                # if the length of our list is less than or equal to N, run the function again
                # ultimately, our final length of list will equal the input N
                recursive(lst, counter)
    
        # run the recursive function 
        recursive(store, N)

        # return the final element in our store list
        return store[-1]